endpoint/platform/legato-target/impl.c

```diff
@@ -17,30 +17,47 @@
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
+#include "endpoint/cipher.h"
 
 #include "legato.h"
 
 #include "interfaces.h"
 
-#define MAXLINE 1024
+static le_sim_Id_t SimId;
 
-#define CBC_IV_SIZE 16
-#define IMSI_LEN 15
-#define READ_BUFFER_SIZE 32
-#define DEFAULT_PORT "/dev/ttyHS0"
+status_t set_device_key(const char *key) {
+  if (key == NULL) {
+    LE_ERROR("Failed to get key");
+    return SC_ENDPOINT_SET_KEY_ERROR;
+  }
 
-static le_sim_Id_t SimId;
+  uint8_t private_key[AES_CBC_KEY_SIZE] = {};
+  memcpy(private_key, key, AES_CBC_KEY_SIZE);
+
+  le_result_t result = secStoreGlobal_Write(ENDPOINT_DEVICE_KEY_NAME, private_key, AES_CBC_KEY_SIZE);
+
+  if (result != LE_OK) {
+    LE_ERROR("Failed to set private key");
+    return SC_ENDPOINT_SET_KEY_ERROR;
+  }
+  return SC_OK;
+}
 
 status_t get_device_key(uint8_t *key) {
-  // FIXME:Need to implement the private key generate algorithm
   if (key == NULL) {
     LE_ERROR("Failed to get key");
     return SC_ENDPOINT_GET_KEY_ERROR;
   }
 
-  char *private_key = "AAAAAAAAAAAAAAAA";
-  memcpy(key, private_key, CBC_IV_SIZE);
+  size_t key_size = AES_CBC_KEY_SIZE;
+  uint8_t private_key[AES_CBC_KEY_SIZE] = {};
 
+  le_result_t result = secStoreGlobal_Read(ENDPOINT_DEVICE_KEY_NAME, private_key, &key_size);
```

Comment:
Since secure storage is used, you have to perform the exception handling according to Legato framework.